### ISSUE TITLE
Documentation example of lint warnings broken for rustc lints

### ISSUE THREAD

**Issue by klensy** [rust-lang/rust#119010]
Rustc lints have placesholder `{{produces}}` in examples that should generate actual warnings for documentation, but it currently not working, as can be seen on https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lint/let_underscore/static.LET_UNDERSCORE_DROP.html, for example; it should contain actual warning text instead of `{{produces}}`.

As of version presented on site:
```
1.76.0-nightly (a96d57bdb 2023-12-15)
```

**Comment by klensy** [comment]
Nope, looks like it works for lint listing, for example https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#lossy-provenance-casts, but not for rustdocs.

**Comment by Nilstrieb** [comment]
I don't think we expect it to actually work for rustdoc?

**Comment by klensy** [comment]
Probably, but it still looks not good.

**Comment by Nilstrieb** [comment]
doing this nicely sounds like quite some nontrivial effort, which I doubt is worth it. keeping it open with these labels anyways.
